File: VB/MvcCustomTextForInternalDashboardErrors/RestrictedControllerFactory.vb
Imports System.Web
Imports System.Web.Mvc

Public Class RestrictedControllerFactory
    Inherits DefaultControllerFactory

    Public Overrides Function CreateController(ByVal requestContext As System.Web.Routing.RequestContext, ByVal controllerName As String) As IController
        Dim controller = MyBase.CreateController(requestContext, controllerName)
        If DirectCast(controller, Object).GetType() Is GetType(DevExpress.DashboardWeb.Mvc.DashboardController) Then
            Throw New HttpException(404, String.Format("The controller '{0}' was not found", controllerName))
        End If
        Return controller
    End Function
End Class

File: CS/MvcCustomTextForInternalDashboardErrors/RestrictedControllerFactory.cs
using System.Web;
using System.Web.Mvc;

namespace MvcCustomTextForInternalDashboardErrors {
    public class RestrictedControllerFactory : DefaultControllerFactory {
        public override IController CreateController(System.Web.Routing.RequestContext requestContext, string controllerName) {
            var controller = base.CreateController(requestContext, controllerName);
            if (controller.GetType() == typeof(DevExpress.DashboardWeb.Mvc.DashboardController)) {
                throw new HttpException(404, string.Format("The controller '{0}' was not found", controllerName));
            }
            return controller;
        }
    }
}